asab/application.py

```diff
@@ -194,6 +199,28 @@ def handler(type):
 		# Every 10 minutes listen for housekeeping
 		self.PubSub.subscribe("Application.tick/600!", self._on_housekeeping_tick)
 
+		# Run the watchdog to detect lost interactivity on the event loop
+		thread = threading.Thread(target=self._run_watchdog_for_event_loop)
+		thread.daemon = True  # Daemonize the thread to ensure it exits with the main program
+		thread.start()
+
+
+	def _run_watchdog_for_event_loop(self):
+		"""
+		Periodically checks the loop interactivity.
+		"""
+
+		while True:
+			current_time = time.time()
+
+			if (current_time - self.LastOnTick60) < (15 * 60):  # Fifteen minute threshold (15 * expected cycle)
+				time.sleep(60)  # Sleep one minute (one cycle)
+				continue
+
+			# The loop lost its interactivity
+			L.critical("The event loop lost its interactivity. Stopping the application!")
+			os._exit(5)  # Cannot use sys.exit inside the thread
```

Comment:
Why?

Reply:
Because sys.exit simply raises SystemExit, which basically terminates the thread but not the application.

Reply:
consider killing the process then using: `os.kill(os.getpid(), signal.SIGKILL)`